### ISSUE TITLE
all error in output

### ISSUE THREAD

**Issue by eidng8** [codeclimate/codeclimate-duplication#156]
I've got errors on every build of my PHP project:

```text
OpenJDK 64-Bit Server VM warning: Using the ParNew young collector with the Serial old collector is deprecated and will likely be removed in a future release
Skipping file ./src/Wiki/Template.php due to exception (CC::Engine::Analyzers::ParserError): `php /usr/src/app/vendor/php-parser/parser.php` exited with code 1:
Parse Error: Syntax error, unexpected T_CONST, expecting T_FUNCTION on line 20

/usr/src/app/lib/cc/engine/analyzers/command_line_runner.rb:24:in `block in run'
org/jruby/ext/timeout/Timeout.java:117:in `timeout'
/usr/src/app/lib/cc/engine/analyzers/command_line_runner.rb:16:in `run'
/usr/src/app/lib/cc/engine/analyzers/php/parser.rb:20:in `parse'
/usr/src/app/lib/cc/engine/analyzers/php/main.rb:24:in `process_file'
/usr/src/app/lib/cc/engine/analyzers/analyzer_base.rb:26:in `run'
/usr/src/app/lib/cc/engine/analyzers/reporter.rb:38:in `block in process_files'
/usr/src/app/lib/cc/engine/analyzers/file_thread_pool.rb:22:in `block in run'
Skipping file ./src/Wiki/Models/Mission.php due to exception (CC::Engine::Analyzers::ParserError): `php /usr/src/app/vendor/php-parser/parser.php` exited with code 1:
Parse Error: Syntax error, unexpected T_CONST, expecting T_FUNCTION on line 22

/usr/src/app/lib/cc/engine/analyzers/command_line_runner.rb:24:in `block in run'
org/jruby/ext/timeout/Timeout.java:117:in `timeout'
/usr/src/app/lib/cc/engine/analyzers/command_line_runner.rb:16:in `run'
/usr/src/app/lib/cc/engine/analyzers/php/parser.rb:20:in `parse'
/usr/src/app/lib/cc/engine/analyzers/php/main.rb:24:in `process_file'
/usr/src/app/lib/cc/engine/analyzers/analyzer_base.rb:26:in `run'
/usr/src/app/lib/cc/engine/analyzers/reporter.rb:38:in `block in process_files'
/usr/src/app/lib/cc/engine/analyzers/file_thread_pool.rb:22:in `block in run'
Skipping file ./src/Wiki/Models/MissionCost.php due to exception (CC::Engine::Analyzers::ParserError): `php /usr/src/app/vendor/php-parser/parser.php` exited with code 1:
Parse Error: Syntax error, unexpected '?' on line 28

/usr/src/app/lib/cc/engine/analyzers/command_line_runner.rb:24:in `block in run'
org/jruby/ext/timeout/Timeout.java:117:in `timeout'
/usr/src/app/lib/cc/engine/analyzers/command_line_runner.rb:16:in `run'
/usr/src/app/lib/cc/engine/analyzers/php/parser.rb:20:in `parse'
/usr/src/app/lib/cc/engine/analyzers/php/main.rb:24:in `process_file'
/usr/src/app/lib/cc/engine/analyzers/analyzer_base.rb:26:in `run'
/usr/src/app/lib/cc/engine/analyzers/reporter.rb:38:in `block in process_files'
/usr/src/app/lib/cc/engine/analyzers/file_thread_pool.rb:22:in `block in run'
Skipping file ./src/Wiki/Models/Skills.php due to exception (CC::Engine::Analyzers::ParserError): `php /usr/src/app/vendor/php-parser/parser.php` exited with code 1:
Parse Error: Syntax error, unexpected T_CONST, expecting T_FUNCTION on line 21

/usr/src/app/lib/cc/engine/analyzers/command_line_runner.rb:24:in `block in run'
org/jruby/ext/timeout/Timeout.java:117:in `timeout'
/usr/src/app/lib/cc/engine/analyzers/command_line_runner.rb:16:in `run'
/usr/src/app/lib/cc/engine/analyzers/php/parser.rb:20:in `parse'
/usr/src/app/lib/cc/engine/analyzers/php/main.rb:24:in `process_file'
/usr/src/app/lib/cc/engine/analyzers/analyzer_base.rb:26:in `run'
/usr/src/app/lib/cc/engine/analyzers/reporter.rb:38:in `block in process_files'
/usr/src/app/lib/cc/engine/analyzers/file_thread_pool.rb:22:in `block in run'
Skipping file ./src/Wiki/Models/CrewMember.php due to exception (CC::Engine::Analyzers::ParserError): `php /usr/src/app/vendor/php-parser/parser.php` exited with code 1:
Parse Error: Syntax error, unexpected '?', expecting T_VARIABLE on line 154

/usr/src/app/lib/cc/engine/analyzers/command_line_runner.rb:24:in `block in run'
org/jruby/ext/timeout/Timeout.java:117:in `timeout'
/usr/src/app/lib/cc/engine/analyzers/command_line_runner.rb:16:in `run'
/usr/src/app/lib/cc/engine/analyzers/php/parser.rb:20:in `parse'
/usr/src/app/lib/cc/engine/analyzers/php/main.rb:24:in `process_file'
/usr/src/app/lib/cc/engine/analyzers/analyzer_base.rb:26:in `run'
/usr/src/app/lib/cc/engine/analyzers/reporter.rb:38:in `block in process_files'
/usr/src/app/lib/cc/engine/analyzers/file_thread_pool.rb:22:in `block in run'
Skipping file ./src/Wiki/Templates/MissionList.php due to exception (CC::Engine::Analyzers::ParserError): `php /usr/src/app/vendor/php-parser/parser.php` exited with code 1:
Parse Error: Syntax error, unexpected '?' on line 98

/usr/src/app/lib/cc/engine/analyzers/command_line_runner.rb:24:in `block in run'
org/jruby/ext/timeout/Timeout.java:117:in `timeout'
/usr/src/app/lib/cc/engine/analyzers/command_line_runner.rb:16:in `run'
/usr/src/app/lib/cc/engine/analyzers/php/parser.rb:20:in `parse'
/usr/src/app/lib/cc/engine/analyzers/php/main.rb:24:in `process_file'
/usr/src/app/lib/cc/engine/analyzers/analyzer_base.rb:26:in `run'
/usr/src/app/lib/cc/engine/analyzers/reporter.rb:38:in `block in process_files'
/usr/src/app/lib/cc/engine/analyzers/file_thread_pool.rb:22:in `block in run'
Skipping file ./src/Wiki/Templates/CrewList.php due to exception (CC::Engine::Analyzers::ParserError): `php /usr/src/app/vendor/php-parser/parser.php` exited with code 1:
Parse Error: Syntax error, unexpected '?' on line 219

/usr/src/app/lib/cc/engine/analyzers/command_line_runner.rb:24:in `block in run'
org/jruby/ext/timeout/Timeout.java:117:in `timeout'
/usr/src/app/lib/cc/engine/analyzers/command_line_runner.rb:16:in `run'
/usr/src/app/lib/cc/engine/analyzers/php/parser.rb:20:in `parse'
/usr/src/app/lib/cc/engine/analyzers/php/main.rb:24:in `process_file'
/usr/src/app/lib/cc/engine/analyzers/analyzer_base.rb:26:in `run'
/usr/src/app/lib/cc/engine/analyzers/reporter.rb:38:in `block in process_files'
/usr/src/app/lib/cc/engine/analyzers/file_thread_pool.rb:22:in `block in run'
Skipping file ./src/Wiki/Templates/Mission.php due to exception (CC::Engine::Analyzers::ParserError): `php /usr/src/app/vendor/php-parser/parser.php` exited with code 1:
Parse Error: Syntax error, unexpected '|', expecting T_VARIABLE on line 198

/usr/src/app/lib/cc/engine/analyzers/command_line_runner.rb:24:in `block in run'
org/jruby/ext/timeout/Timeout.java:117:in `timeout'
/usr/src/app/lib/cc/engine/analyzers/command_line_runner.rb:16:in `run'
/usr/src/app/lib/cc/engine/analyzers/php/parser.rb:20:in `parse'
/usr/src/app/lib/cc/engine/analyzers/php/main.rb:24:in `process_file'
/usr/src/app/lib/cc/engine/analyzers/analyzer_base.rb:26:in `run'
/usr/src/app/lib/cc/engine/analyzers/reporter.rb:38:in `block in process_files'
/usr/src/app/lib/cc/engine/analyzers/file_thread_pool.rb:22:in `block in run'
Skipping file ./src/Wiki/Templates/MissionCost.php due to exception (CC::Engine::Analyzers::ParserError): `php /usr/src/app/vendor/php-parser/parser.php` exited with code 1:
Parse Error: Syntax error, unexpected T_CONST, expecting T_FUNCTION on line 29

/usr/src/app/lib/cc/engine/analyzers/command_line_runner.rb:24:in `block in run'
org/jruby/ext/timeout/Timeout.java:117:in `timeout'
/usr/src/app/lib/cc/engine/analyzers/command_line_runner.rb:16:in `run'
/usr/src/app/lib/cc/engine/analyzers/php/parser.rb:20:in `parse'
/usr/src/app/lib/cc/engine/analyzers/php/main.rb:24:in `process_file'
/usr/src/app/lib/cc/engine/analyzers/analyzer_base.rb:26:in `run'
/usr/src/app/lib/cc/engine/analyzers/reporter.rb:38:in `block in process_files'
/usr/src/app/lib/cc/engine/analyzers/file_thread_pool.rb:22:in `block in run'
Skipping file ./src/Wiki/Api/ExpandTemplates.php due to exception (CC::Engine::Analyzers::ParserError): `php /usr/src/app/vendor/php-parser/parser.php` exited with code 1:
Parse Error: Syntax error, unexpected '?' on line 63

/usr/src/app/lib/cc/engine/analyzers/command_line_runner.rb:24:in `block in run'
org/jruby/ext/timeout/Timeout.java:117:in `timeout'
/usr/src/app/lib/cc/engine/analyzers/command_line_runner.rb:16:in `run'
/usr/src/app/lib/cc/engine/analyzers/php/parser.rb:20:in `parse'
/usr/src/app/lib/cc/engine/analyzers/php/main.rb:24:in `process_file'
/usr/src/app/lib/cc/engine/analyzers/analyzer_base.rb:26:in `run'
/usr/src/app/lib/cc/engine/analyzers/reporter.rb:38:in `block in process_files'
/usr/src/app/lib/cc/engine/analyzers/file_thread_pool.rb:22:in `block in run'
Skipping file ./src/Wiki/Api/Parse.php due to exception (CC::Engine::Analyzers::ParserError): `php /usr/src/app/vendor/php-parser/parser.php` exited with code 1:
Parse Error: Syntax error, unexpected '?' on line 204

/usr/src/app/lib/cc/engine/analyzers/command_line_runner.rb:24:in `block in run'
org/jruby/ext/timeout/Timeout.java:117:in `timeout'
/usr/src/app/lib/cc/engine/analyzers/command_line_runner.rb:16:in `run'
/usr/src/app/lib/cc/engine/analyzers/php/parser.rb:20:in `parse'
/usr/src/app/lib/cc/engine/analyzers/php/main.rb:24:in `process_file'
/usr/src/app/lib/cc/engine/analyzers/analyzer_base.rb:26:in `run'
/usr/src/app/lib/cc/engine/analyzers/reporter.rb:38:in `block in process_files'
/usr/src/app/lib/cc/engine/analyzers/file_thread_pool.rb:22:in `block in run'
```

**Comment by maxjacobson** [comment]
@eidng8 My interpretation of this error message is that our php parser thinks your code has a syntax error. We support PHP 5 and PHP 7 syntax. Which version are you using? Have you verified that there are no syntax errors in the files mentioned? e.g. `./src/Wiki/Templates/MissionCost.php`

**Comment by eidng8** [comment]
I'm using PHP7.1, everything are linted before commit.

**Comment by maxjacobson** [comment]
@eidng8  Just shipped #177, which adds support for 7.1 syntax. Will you try creating a new build?

**Comment by eidng8** [comment]
A lot better, but still has something:

https://codeclimate.com/github/eidng8/sttbot/builds/167

**Comment by maxjacobson** [comment]
Referring to the "MD5" warning? That something that's always printed. We should probably update that to only print if the engine has opted into debug information.

**Comment by dblandin** [comment]
Looks like this has been resolved with #177. Going to go ahead and close this issue.
